source/overloading.tex

```diff
@@ -3657,60 +3649,42 @@
 \indextext{overloading!function call operator}
 
 \pnum
-\tcode{operator()}
-shall be a non-static member function with an arbitrary number of
-parameters.
-It can have default arguments.
-It implements the function call syntax
+A \defnadj{function call}{operator function}
+is a function named \tcode{\keyword{operator}()}
+that is a non-static member function with an arbitrary number of parameters.
+It may have default arguments.
+For an expression of the form
 \begin{ncsimplebnf}
 postfix-expression \terminal{(} \opt{expression-list} \terminal{)}
 \end{ncsimplebnf}
-where the
-\grammarterm{postfix-expression}
-evaluates to a class object and the possibly empty
-\grammarterm{expression-list}
-matches the parameter list of an
-\tcode{operator()}
-member function of the class.
-Thus, a call
-\tcode{x(arg1,...)}
-is interpreted as
-\tcode{x.op\-er\-a\-tor()(arg1, ...)}
-for a class object
-\tcode{x}
-of type
-\tcode{T}
-if
-\tcode{T::operator()(T1,}
-\tcode{T2,}
-\tcode{T3)}
-exists and if the operator is selected as the best match function by
-the overload resolution mechanism\iref{over.match.best}.
+where the \grammarterm{postfix-expression} evaluates to a class object,
+the operator function
+is selected by overload resolution\iref{over.match.oper},
+and the expression is interpreted as
+\begin{ncsimplebnf}
+postfix-expression \terminal{.} \keyword{operator} \terminal{(}\terminal{)} \terminal{(} \opt{expression-list} \terminal{)}
+\end{ncsimplebnf}
```

Comment:
This is wrong for the case where we select a surrogate candidate. The old wording was correct but incomplete.

Reply:
We use over.call.object, not over.match.oper for the function call operator. Fixed xref.

Reply:
Fixed, also removing the bad-placed "how do we call this in the end" paragraph from over.call.object.